activerecord/lib/active_record/model_schema.rb

```diff
@@ -432,6 +432,7 @@ def reset_column_information
         connection.schema_cache.clear_data_source_cache!(table_name)
 
         reload_schema_from_cache
+        initialize_find_by_cache
```

Comment:
/cc @tenderlove Make sense to clear find-by cache when we reset_column_information?